control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -360,7 +360,15 @@ func (r *Reconciler) finalizeKind(ctx context.Context, broker *eventing.Broker)
 	if !externalTopic {
 		topicConfig, err := r.topicConfig(logger, broker, brokerConfig)
 		if err != nil {
-			return fmt.Errorf("failed to resolve broker config: %w", err)
+
+			// On finalize, we fail to get a valid config, we can safely ignore and return nil
+			// no further actions are needed since we are also not putting the finalizer on given secret
+			// if we are not having a valid topic config
+			if strings.Contains(err.Error(), "validating topic config") {
+				return nil
```

Comment:
@matzew I don't think this commit addresses the discussion we had in 2022.

See https://github.com/knative-sandbox/eventing-kafka-broker/pull/2853#issuecomment-1363998260.

Basically, here, when we return nil, we keep the finalizer string still kept on the secret because line 399 (`removeFinalizerSecret`) is never called. However, we should remove the finalizer string on the secret, even when topic config is wrong.

Reply:
There is no finalizer on a secret, if we have "invalid topic config".

I guess I can update the test to have some secret there

Reply:
On the other hand, the finalizer is not set at all on the secret in `reconcileKind`, if the topicConfig is wrong...

/lgtm